File: app/src/main/java/com/absinthe/kage/device/cmd/PlayPreviousCommand.java
package com.absinthe.kage.device.cmd;

import com.absinthe.kage.connect.protocol.IpMessageConst;
import com.absinthe.kage.device.Command;
import com.absinthe.kage.device.CommandBuilder;
import com.absinthe.kage.device.client.Client;

public class PlayPreviousCommand extends Command {

    public static final String MESSAGE = "PLAY_PRE";

    public PlayPreviousCommand() {
        cmd = IpMessageConst.MEDIA_PLAY_PREVIOUS;
    }

    @Override
    public String pack() {
        return new CommandBuilder()
                .with(this)
                .append(MESSAGE)
                .build();
    }

    @Override
    public void doWork(Client client, String received) {

    }

    @Override
    public boolean parseReceived(String received) {
        return false;
    }
}

File: app/src/main/kotlin/com/absinthe/kage/device/cmd/PlayPreviousCommand.kt
package com.absinthe.kage.device.cmd

import com.absinthe.kage.connect.protocol.IpMessageConst
import com.absinthe.kage.device.Command
import com.absinthe.kage.device.CommandBuilder
import com.absinthe.kage.device.client.Client

class PlayPreviousCommand : Command() {

    override fun pack(): String {
        return CommandBuilder()
                .with(this)
                .append(MESSAGE)
                .build()
    }

    override fun doWork(client: Client, received: String) {}

    override fun parseReceived(received: String): Boolean {
        return false
    }

    companion object {
        const val MESSAGE = "PLAY_PRE"
    }

    init {
        cmd = IpMessageConst.MEDIA_PLAY_PREVIOUS
    }
}